src/libsyntax_ext/format.rs

```diff
@@ -996,9 +996,10 @@ pub fn expand_preparsed_format_args(ecx: &mut ExtCtxt,
                         ));
                     }
                     if suggestions.len() > 0 {
-                        diag.multipart_suggestion(
-                            "format specifiers use curly braces",
-                            suggestions,
+                        diag.multipart_suggestion_with_applicability(
+                                                    "format specifiers use curly braces",
+                                                    suggestions,
+                                                    Applicability::MachineApplicable,
```

Comment:
I love this fix. This is just magic. I hope it actually works each time it yields suggestions :D